### PR TITLE
better support for stages

### PR DESCRIPTION
**Pull request by DannyDouglass**
@TimothyJones not sure what the process is here, but this pr introduces an intentional api gateway that allows us to target custom stages and not generate a random stage named `Stage`. these changes are deployed in a development environment and tested against that with success (minus github 2fa users as discussed in #24).

image

### CODE REVIEW

File: scripts/deploy.sh [hunk 1]
```diff
@@ -20,5 +20,5 @@ source ./config.sh
 
 OUTPUT_TEMPLATE_FILE="$PROJECT_ROOT/serverless-output.yml"
 aws s3 mb "s3://$BUCKET_NAME" --region "$REGION" || true
+sam deploy --region "$REGION" --template-file "$OUTPUT_TEMPLATE_FILE" --stack-name "$STACK_NAME" --parameter-overrides GitHubClientIdParameter="$GITHUB_CLIENT_ID" GitHubClientSecretParameter="$GITHUB_CLIENT_SECRET" CognitoRedirectUriParameter="$COGNITO_REDIRECT_URI" StageNameParameter="$STAGE_NAME" --capabilities CAPABILITY_IAM
```
Comment (DannyDouglass) [inline scripts/deploy.sh]: pass in the `stage`

File: template.yml [hunk 2]
```diff
@@ -36,8 +36,15 @@ Parameters:
...
     MinLength: 1
+  StageNameParameter:
+    Type: String
 
 Resources:
+  GithubOAuthApi:
+    Type: AWS::Serverless::Api
+    Properties:
+      StageName: !Ref StageNameParameter
+      OpenApiVersion: "2.0"
```
Comment (DannyDouglass) [inline template.yml]: needed to get around the `sam` stage bug: https://github.com/awslabs/serverless-application-model/issues/191

**Comment by TimothyJones** [comment]
Thanks very much for this! I especially appreciate the documentation updates 🙏

**Comment by DannyDouglass** [comment]
happy to help @TimothyJones!

**Pull request merged**